### PR TITLE
Add new submodule MMseqs/taxonomy

### LINKED ISSUE #4970: new module: MMSEQS/TAXONOMY
### Is there an existing module for this?

- [X] I have searched for the existing module

### Is there an open PR for this?

- [X] I have searched for existing PRs

### Is there an open issue for this?

- [X] I have searched for existing issues

### Are you going to work on this?

- [X] If I'm planning to work on this module, I added myself to the `Assignees` to facilitate tracking who is working on the module

### PR DESCRIPTION
**Pull request by Darcy220606**
## PR checklist

Closes #4970

- [x] This comment contains a description of changes (with reason).
- [x] If you've fixed a bug or added code that should be tested, add tests!
- [x] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [x] If necessary, include test data in your PR.
- [x] Remove all TODO statements.
- [x] Emit the `versions.yml` file.
- [x] Follow the naming conventions.
- [x] Follow the parameters requirements.
- [x] Follow the input/output options guidelines.
- [x] Add a resource `label`
- [x] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
  - For modules:
    - [x] `nf-core modules test <MODULE> --profile docker`
    - [x] `nf-core modules test <MODULE> --profile singularity`
    - [x] `nf-core modules test <MODULE> --profile conda`
  - For subworkflows:
    - [ ] `nf-core subworkflows test <SUBWORKFLOW> --profile docker`
    - [ ] `nf-core subworkflows test <SUBWORKFLOW> --profile singularity`
    - [ ] `nf-core subworkflows test <SUBWORKFLOW> --profile conda`

### CODE REVIEW

File: modules/nf-core/mmseqs/taxonomy/tests/main.nf.test.snap [hunk 1]
```diff
@@ -0,0 +1,18 @@
+{
+    "taxonomy - bacteroides_fragilis - genome_nt": {
+        "content": [
+            [
+                "test_query_taxonomy"
+            ]
+        ],
+        "timestamp": "2024-02-23T15:31:53.321266978"
```
Comment (jasmezz) [inline modules/nf-core/mmseqs/taxonomy/tests/main.nf.test.snap:8]: Interesting that there is no `versions.yml` mentioned although you check for it in main.nf.test. I don't understand that, but apparently the tests pass :shrug:

**Review by jasmezz — CHANGES_REQUESTED** [review]
Looks great overall :tada: Just 1 major question for the `main.nf.test` plus a few minor tweaks.

File: modules/nf-core/mmseqs/taxonomy/meta.yml [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+      description: an MMseqs2 database with target data including the taxonomy classification
+output:
+  - meta:
+      type: map
+      description: |
+        Groovy Map containing sample information
+        e.g. `[ id:'test', single_end:false ]`
+  - db_taxonomy:
+      type: directory
+      description: The created MMseqs2 database with taxonomic classifications
```
[the remark this answers is not in the document]
Comment (Darcy220606) [reply modules/nf-core/mmseqs/taxonomy/meta.yml:45]: its a dtabase in this case, which stilln eeds to be converted to a table with another module

File: modules/nf-core/mmseqs/taxonomy/tests/main.nf.test [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+                input[0] = MMSEQS_CREATEDB.out.db
+                input[1] = MMSEQS_DATABASES.out.database
+                """
+            }
+        }
+
+        then {
+            assertAll(
+                { assert process.success },
+                { assert snapshot(process.out.db_taxonomy.collect { file(it[1]).getName() }).match() },
```
[the remark this answers is not in the document]
Comment (Darcy220606) [reply modules/nf-core/mmseqs/taxonomy/tests/main.nf.test:50]: Yes i checked that, but there was always 'Empty file with md5sum' error coming up when i linted the module. A way around this is to to check for the ouput names instead.

[the remark this answers is not in the document]
Comment (Darcy220606) [reply modules/nf-core/mmseqs/taxonomy/tests/main.nf.test:50]: ok changed it now to output the file names instead

File: modules/nf-core/mmseqs/taxonomy/tests/main.nf.test [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+                input[1] = MMSEQS_DATABASES.out.database
+                """
+            }
+        }
+
+        then {
+            assertAll(
+                { assert process.success },
+                { assert snapshot(process.out.db_taxonomy.collect { file(it[1]).getName() }).match()},
+                { assert process.out.versions }
```
[the remark this answers is not in the document]
Comment (Darcy220606) [reply modules/nf-core/mmseqs/taxonomy/tests/main.nf.test:73]: That doesnt work unfortunately. Only when i compare teh names does it work for some reason.

**Review by jasmezz — APPROVED** [review]
Looks good! :partying_face: After tests succeeded, you can just add my suggestion with [skip ci]. (Don't trigger tests again because the blank line does not change the code)

File: modules/nf-core/mmseqs/taxonomy/tests/main.nf.test [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+        run("MMSEQS_DATABASES") {
+            script "modules/nf-core/mmseqs/databases/main.nf"
+            process {
+            """
+            input[0] = 'SILVA'
+            """
+            }
+        }
+    }
+    test("mmseqs/taxonomy - bacteroides_fragilis - genome_nt") {
```
[the remark this answers is not in the document]
Comment (Darcy220606) [reply modules/nf-core/mmseqs/taxonomy/tests/main.nf.test:37]: Big thanks @jasmezz for the reviews !!!

**Pull request merged**